cmd/kubeadm/app/util/copy_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+	"testing"
+)
+
+func TestCopyFile(t *testing.T) {
+	tmpdir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Fatal("Failed to create temporary directory")
+	}
+	defer func() {
+		err = os.RemoveAll(tmpdir)
+		if err != nil {
+			t.Fatal("Failed to remove temporary directory")
+		}
+	}()
+	tmpfile, err := os.CreateTemp(tmpdir, "")
+	if err != nil {
+		t.Fatalf("Failed to create temporary file")
+	}
+	restrictedFile, err := os.CreateTemp(tmpdir, "")
+	if err != nil {
+		t.Fatal("Failed to create temporary restricted file")
+	}
+	err = restrictedFile.Chmod(0000)
```

Comment:
@aojea are we still running unit tests as root in k8s? this could explain why this passed in this run here and is failing in other environments